cmd/frontend/graphqlbackend/external_account.go

```diff
@@ -60,11 +60,21 @@ func (r *externalAccountResolver) RefreshURL() *string {
 }
 
 func (r *externalAccountResolver) AccountData(ctx context.Context) (*JSONValue, error) {
-	// 🚨 SECURITY: Only the site admins and the user can view this information.
-	// TODO(jchen): Only allow the authenticated user getting back accountData for
-	// GitHub and GitLab, https://github.com/sourcegraph/sourcegraph/issues/20978
-	if err := backend.CheckSiteAdminOrSameUser(ctx, actor.FromContext(ctx).UID); err != nil {
-		return nil, err
+	// 🚨 SECURITY: It is only safe to assume account data of GitHub and GitLab do
+	// not contain sensitive information that is not known to the user (which is
+	// accessible via APIs by users themselves). We cannot take the same assumption
+	// for other types of external accounts.
+	//
+	// Therefore, the site admins and the user can view account data of GitHub and
+	// GitLab, but only site admins can view account data for all other types.
+	if r.account.ServiceType == extsvc.TypeGitHub || r.account.ServiceType == extsvc.TypeGitLab {
```

Comment:
Does it matter if it's github.com or some other GHE instance? Should we test that too? If so, perhaps we could use https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/extsvc/codehost.go?subtree=true#L44?

Reply:
This code would also be easier to read like this:

```suggestion
  var err error
	if r.account.ServiceType == extsvc.TypeGitHub || r.account.ServiceType == extsvc.TypeGitLab {
	  err = backend.CheckSiteAdminOrSameUser(ctx, actor.FromContext(ctx).UID)
	} else {
	  err = backend.CheckUserIsSiteAdmin(ctx, actor.FromContext(ctx).UID)
	}
	
	if err != nil {
	   return nil, err
	 }
```

Reply:
> Does it matter if it's github.com or some other GHE instance? Should we test that too? If so, perhaps we could use [sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/extsvc/codehost.go?subtree=true#L44](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/extsvc/codehost.go?subtree=true#L44)?

No it doesn't matter.